pylib/anki/collection.py

```diff
@@ -92,6 +92,33 @@ class LegacyCheckpoint:
 LegacyUndoResult = Union[None, LegacyCheckpoint, LegacyReviewUndo]
 
 
+class ExportLimit:
+    """Limit to what will be exported. Either specific notes, or a deck, or the
+    whole collection (if neither is set). Only the last set value is preserved.
+    """
```

Comment:
This is similar to how protobuf oneofs perform, so there is precedent, but I suspect the multiple-wrapper-classes is a bit more Pythonic. Some disadvantages of this approach:

- It requires some more boilerplate, especially as each added case requires updates to the code for the other cases. 
- The implicit clearing of other fields is not normal/intuitive, so it makes it a bit easier to make mistakes.

Reply:
It was an attempt to make this more ergonomic and avoid instance checks, but the fact that I used my own API wrong convinces me, that it failed. 😄 I'll revert it to what you suggested.